src/index.ts

```diff
@@ -372,3 +376,53 @@ export class CachedClient implements ICachedHttpClient {
     return await this._httpClient.delete(route);
   }
 }
+
+export const redact = (options: RequestOptions) => {
+  const clone = copy(options);
+  redactSensitiveHeaders(clone);
+  redactSensitiveProps(clone);
+  return clone;
+};
+
+export const redactSensitiveHeaders = (options: RequestOptions) => {
+  if (options.headers === undefined) return;
+
+  for (const prop of Object.keys(options.headers ?? {})) {
+    for (const propMatch of Sensitive.headers) {
+      if (!propMatch.test(prop)) continue;
+      // eslint-disable-next-line no-param-reassign
+      options.headers[prop] = '<redacted>';
+    }
+  }
+};
+
+export const redactSensitiveProps = (options: RequestOptions) => {
+  let jsonOrForm;
+  if ('json' in options) {
+    jsonOrForm = options.json;
+  } else if ('form' in options) {
+    jsonOrForm = options.form;
+  }
+  if (jsonOrForm === undefined) return;
+
+  for (const prop of Object.keys(jsonOrForm)) {
+    for (const propMatch of Sensitive.props) {
+      if (!propMatch.test(prop)) continue;
+      jsonOrForm[prop] = '<redacted>';
+    }
+  }
+};
+
+const Sensitive = {
+  headers: [/[aA]uthorization/],
+  props: [
+    /[pP]ass(word)?/,
+    /[eE]mail/,
+    /[tT]oken/,
+    /[sS]ecret/,
+    /[cC]lient_?[iI]d/,
+    /[cC]lient_?[sS]ecret/,
+    /[uU]ser(name)?/,
+    /[aA]uthorization/
+  ]
+};
```

Comment:
Maybe just skip regexes and type 'em all out for performance?

Reply:
Probably neglectable,  otherwise convert to all Uppercase